style.md

```diff
@@ -1923,6 +1924,90 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps
+
+Prefer `make(..)` for empty maps, and maps populated
+programmatically. This makes map initialization visually
+distinct from declaration, and it makes it easy to add size
+hints later if available.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+var (
+  m1 = map[T1]T2{}
+  m2 map[T1]T2
+)
+```
+
+</td><td>
+
+```go
+var (
+  m1 = make(map[T1]T2)
+  m2 map[T1]T2
+)
+```
+
+</td></tr>
+<tr><td>
+
+Declaration and initialization are visually similar.
+
+</td><td>
+
+Declaration and initialization are more visually distinct.
+
+</td></tr>
+</tbody></table>
+
+On the other hand, if the map holds a fixed list of elements,
+use map literals to initialize the map.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+m := make(map[T1]T2, 3)
+m[k1] = v1
+m[k2] = v2
+m[k3] = v3
+```
+
+</td><td>
+
+```go
+m := map[T1]T2{
+  k1: v1,
+  k2: v2,
+  k3: v3,
+}
+```
+
+</td></tr>
+</tbody></table>
+
+
+While it's impractical to prescribe a universal strategy for map initialization,
+there are some basic rules of thumb you can use here.
+
+- Use `make()` when...
+  - an empty map is being initialized
+  - elements must be added programatically, versus as part of a literal
+  - when a size hint is available
+  - when the rest of the codebase predominantly uses `make()`
+- Use type literals when...
+  - a fixed set of elements is being added at initialization time
+  - when the rest of the codebase predominantly uses literals
+
+There are more nuances than outlined in this guide; when in doubt,
+[be consistent](#be-consistent).
```

Comment:
This makes sense, but is also quite verbose. I'd prefer to keep recommendations short and succinct so it's easier to parse all the rules. Since we'll have be-consistent as a top-level guideline, I don't think we have to repeat it here.

I think this can be summarized as
```
Use map literals when a fixed set of elements is added at initialization time, otherwise use `make` and specify a size hint if available.

Reply:
modulo adding "when initializing an empty map", sgtm!